### PR TITLE
Update dependencies and fix warnings.

### PR DESCRIPTION
**Pull request by martincarrera**

### CODE REVIEW

File: public/main/main-window.js [hunk 1]
```diff
@@ -12,6 +12,9 @@ const restart = require('./client/electron-restart')
 
 let mainWindow
 
+// Disable electron security warnings since local content is served via http
+process.env.ELECTRON_DISABLE_SECURITY_WARNINGS = true
```
Comment (gabmontes) [inline public/main/main-window.js]: This should be disable only in dev mode. Right?

File: package.json [hunk 2]
```diff
@@ -68,9 +68,10 @@
   "devDependencies": {
     "concurrently": "3.6.0",
     "devtron": "1.4.0",
-    "electron": "1.8.8",
-    "electron-builder": "20.22.0",
+    "electron": "4.0.5",
+    "electron-builder": "20.38.5",
     "electron-devtools-installer": "2.2.4",
+    "electron-rebuild": "1.8.4",
```
Comment (gabmontes) [inline package.json:75]: What is this needed for?

File: package.json [hunk 3]
```diff
@@ -32,13 +32,13 @@
...
-    "electron-is-dev": "0.3.0",
+    "electron-context-menu": "0.11.0",
+    "electron-debug": "2.1.0",
+    "electron-is-dev": "1.0.1",
     "electron-log": "3.0.1",
     "electron-settings": "3.2.0",
-    "electron-unhandled": "1.1.0",
-    "electron-updater": "2.23.3",
+    "electron-unhandled": "2.1.0",
+    "electron-updater": "4.0.6",
```
Comment (gabmontes) [inline package.json:42]: No breaking changes in the update process as current users will still be in 2.x until they upgrade?

File: public/main/main-window.js [hunk 4]
```diff
@@ -80,6 +83,7 @@ function loadWindow () {
     backgroundColor: '#323232',
     webPreferences: {
       nodeIntegration: false,
+      contextIsolation: false,
```
Comment (gabmontes) [inline public/main/main-window.js:88]: Default is `false`. Is this needed?

File: public/main/main-window.js (hunk 1 above)
Comment (martincarrera) [reply public/main/main-window.js]: Security warnings only appear in DEV mode, so this env var doesn't do anything in production mode. However, I will change this to only set the variable in DEV environment.

File: public/main/main-window.js (hunk 4 above)
Comment (pablen) [reply public/main/main-window.js:88]: [Electron team plans to change the defaults for some options in next version (5)](https://electronjs.org/blog/electron-4-0#deprecations) so a warning is thrown in current version if the values of those options are not explicitly defined.

Comment (gabmontes) [reply public/main/main-window.js:88]: So we should set it to `false` explicitly.

**Review by pablen — APPROVED** [review]

**Review by gabmontes — APPROVED** [review]

**Pull request merged**